kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -260,50 +261,87 @@ impl<'tcx> GotocCtx<'tcx> {
+                        *field,
+                        field_ty_or_var.expect_type(),
+                    )),
                     // if we fall here, then we are handling either a struct or a union
                     ty::Adt(def, _) => {
-                        let field = &def.variants().raw[0].fields[*f];
-                        Ok(res.member(&field.name.to_string(), &self.symbol_table))
+                        let field = &def.variants().raw[0].fields[*field];
+                        Ok(parent_expr.member(&field.name.to_string(), &self.symbol_table))
+                    }
+                    ty::Closure(..) => {
+                        Ok(parent_expr.member(&field.index().to_string(), &self.symbol_table))
                     }
-                    ty::Closure(..) => Ok(res.member(&f.index().to_string(), &self.symbol_table)),
                     ty::Generator(..) => {
-                        let field_name = self.generator_field_name(f.as_usize());
-                        Ok(res
+                        let field_name = self.generator_field_name(field.as_usize());
+                        Ok(parent_expr
                             .member("direct_fields", &self.symbol_table)
                             .member(field_name, &self.symbol_table))
                     }
                     _ => unimplemented!(),
                 }
             }
             // if we fall here, then we are handling an enum
-            TypeOrVariant::Variant(v) => {
-                let field = &v.fields[*f];
-                Ok(res.member(&field.name.to_string(), &self.symbol_table))
+            TypeOrVariant::Variant(parent_var) => {
+                let field = &parent_var.fields[*field];
+                Ok(parent_expr.member(&field.name.to_string(), &self.symbol_table))
             }
             TypeOrVariant::GeneratorVariant(_var_idx) => {
-                let field_name = self.generator_field_name(f.index());
-                Ok(res.member(field_name, &self.symbol_table))
+                let field_name = self.generator_field_name(field.index());
+                Ok(parent_expr.member(field_name, &self.symbol_table))
             }
         }
     }
 
+    /// This is a SIMD vector, which has 2 possible internal representations:
+    /// 1- Multi-field representation (original and currently deprecated)
+    ///    In this case, a field is one lane (i.e.: one element)
+    ///    Example:
+    /// ```ignore
+    ///    pub struct i64x2(i64, i64);
+    ///    fn main() {
+    ///      let v = i64x2(1, 2);
+    ///      assert!(v.0 == 1); // refers to the first i64
+    ///      assert!(v.1 == 2);
+    ///    }
+    /// ```
+    /// 2- Array-based representation
+    ///    In this case, the projection refers to the entire array.
+    /// ```ignore
+    ///    pub struct i64x2([i64; 2]);
+    ///    fn main() {
+    ///      let v = i64x2([1, 2]);
+    ///      assert!(v.0 == [1, 2]); // refers to the entire array
```

Comment:
Ok, it refers to the entire array. But what does `v.1` refer to then? Can you add that case in this example?

Reply:
There is no `v.1`. it's a compilation error.